File: java/demo/Ice/callback/CallbackReceiverI.java
// **********************************************************************
//
// Copyright (c) 2003 - 2004
// ZeroC, Inc.
// North Palm Beach, FL, USA
//
// All Rights Reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************


public final class CallbackReceiverI extends CallbackReceiver
{
    public void
    callback(Ice.Current current)
    {
        System.out.println("received callback");
    }
}

File: cs/demo/Ice/sl/bidir/bidir/CallbackReceiverI.cs
// **********************************************************************
//
// Copyright (c) 2003-2014 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using Demo;

namespace bidir
{
    public class CallbackReceiverI : CallbackReceiverDisp_
    {
        public CallbackReceiverI(MainPage page)
        {
            _page = page;
        }

        public override void callback(int num, Ice.Current current)
        {
            _page.appendText("received callback #" + num.ToString() + Environment.NewLine);
        }

        private MainPage _page;
    }
}
